lnwallet/witnessgen.go

```diff
@@ -15,16 +15,16 @@ type WitnessType uint16
 const (
 	// CommitmentTimeLock is a witness that allows us to spend the output of a
 	// commitment transaction after a relative lock-time lockout.
-	CommitmentTimeLock WitnessType = 0
+	CommitmentTimeLock WitnessType = iota
```

Comment:
With this change, it's possible that a later movement or shift int he ordering of the declared `WitnessType` values could result in a breaking database change (and a possible inability to properly sweep outputs). This is due to the fact that the persistence code uses the witness type directly in the on-disk serialization of the contested outputs.  

To avoid a situation like this, we either need to revert this change, or use a distinct mapping with the database code itself. Personally, I have the latter option as it further isolates the daemon's database serialization from a define type in `lnwallet`.